### ISSUE TITLE
NPE

### ISSUE THREAD

**Issue by digitalheir** [google/ksp#439]
Version 1.5.0-1.0.0-alpha09

```
e: java.lang.NullPointerException
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl$simpleName$2.invoke(KSFunctionDeclarationImpl.kt:50)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl$simpleName$2.invoke(KSFunctionDeclarationImpl.kt:46)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl.getSimpleName(KSFunctionDeclarationImpl.kt:46)
	at com.google.devtools.ksp.symbolCollector.visitDeclaration(Incremental.kt:144)
	at com.google.devtools.ksp.symbolCollector.visitDeclaration(Incremental.kt:137)
	at com.google.devtools.ksp.visitor.KSDefaultVisitor.visitFunctionDeclaration(KSAllSupersVisitor.kt:39)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl.accept(KSFunctionDeclarationImpl.kt:104)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:155)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:137)
	at com.google.devtools.ksp.visitor.KSDefaultVisitor.visitFile(KSAllSupersVisitor.kt:34)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFileImpl.accept(KSFileImpl.kt:59)
	at com.google.devtools.ksp.IncrementalContext.collectDefinedSymbols(Incremental.kt:215)
	at com.google.devtools.ksp.IncrementalContext.calcDirtyFiles(Incremental.kt:402)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:128)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:119)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:85)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:514)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:505)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:112)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:505)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:189)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:155)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:169)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:52)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:88)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:44)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:98)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1575)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:359)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:562)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:796)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:677)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:676)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Build: AI-202.7660.26.42.7351085, 202105101744, 

AI-202.7660.26.42.7351085, JRE 11.0.8+0-b944-P17168821x64 JetBrains s.r.o, OS Linux(amd64) v5.8.0-53-generic, screens 3840x2160

AS: 4.2.1; Kotlin plugin: 202-1.5.0-release-764-AS8194.7; Android Gradle Plugin: 4.2.1; Gradle: 7.0; NDK: from local.properties: (not specified), latest from SDK: (not found); LLDB: pinned revision 3.1 not found, latest from SDK: (package not found); CMake: from local.properties: (not specified), latest from SDK: 3.18.1-g262b901, from PATH: (not found)Source: user_sentiment_feedback

**Comment by trevjonez** [comment]
Same/similar issue in 1.5.20-1.0.0-beta04

Slightly different stack. going to attach the debugger and see if I can figure out what the trigger is.
```
e: java.lang.NullPointerException
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl$simpleName$2.invoke(KSFunctionDeclarationImpl.kt:50)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl$simpleName$2.invoke(KSFunctionDeclarationImpl.kt:46)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFunctionDeclarationImpl.getSimpleName(KSFunctionDeclarationImpl.kt:46)
	at com.google.devtools.ksp.UtilsKt.isConstructor(utils.kt:267)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSClassDeclarationImpl$declarations$2.invoke(KSClassDeclarationImpl.kt:86)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSClassDeclarationImpl$declarations$2.invoke(KSClassDeclarationImpl.kt:70)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSClassDeclarationImpl.getDeclarations(KSClassDeclarationImpl.kt:70)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:154)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:137)
	at com.google.devtools.ksp.visitor.KSDefaultVisitor.visitClassDeclaration(KSAllSupersVisitor.kt:82)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSClassDeclarationImpl.accept(KSClassDeclarationImpl.kt:122)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:155)
	at com.google.devtools.ksp.symbolCollector.visitDeclarationContainer(Incremental.kt:137)
	at com.google.devtools.ksp.visitor.KSDefaultVisitor.visitFile(KSAllSupersVisitor.kt:34)
	at com.google.devtools.ksp.symbol.impl.kotlin.KSFileImpl.accept(KSFileImpl.kt:60)
	at com.google.devtools.ksp.IncrementalContext.collectDefinedSymbols(Incremental.kt:215)
	at com.google.devtools.ksp.IncrementalContext.registerGeneratedFiles(Incremental.kt:519)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:133)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:120)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:86)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:540)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:531)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:113)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:531)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.repeatAnalysisIfNeeded(KotlinToJVMBytecodeCompiler.kt:501)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:188)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:154)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:169)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:52)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:90)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:44)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:98)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1575)
	at jdk.internal.reflect.GeneratedMethodAccessor109.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:359)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:562)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:796)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:677)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:676)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by trevjonez** [comment]
Looks like some of my generated out put produced some bad formatting.
```kotlin
public class MembershipResourceIdJsonAdapter : JsonAdapter<MembershipResourceId>() {
  public override fun toJson(writer: JsonWriter, `value`: MembershipResourceId?): Unit {
    writer.value(`value`?.`value`)
  }

  public override fun fromJson(reader: JsonReader): MembershipResourceId? = reader.nextString()?.let
      { MembershipResourceId(it) }
}
```

`fromJson` expression wrapping where it did is what causes the issue it seems.

**Comment by ting-yuan** [comment]
Thank you @trevjonez, the test case is really useful! I'm able to shrink it down to a top level or member lambda, which is not a valid declaration:
```
{}
```

A function literal in wrong place also produces the same error:
```
fun() = 1
```

We'll transform this exception into a meaningful error.

**Comment by ting-yuan** [comment]
Fixed in #549
